wizeline/wize-q-remix
# feat: Poc/postgresql

#### What does this PR do?
- Migrates from MySQL to PostgreSQL
- update the Prisma schema 
- update the Components
- update the controller 
- update the tests
- add the files necessaries to create a new db in PostgreSQL and migrate the data. 


#### How should this be manually tested?
1. Include PostgreSQL in the docker container [the configuration is on **_docker-compose.yaml_**] 
           run de command `docker-compose up -d`
2. Download the file '_dump_wizeq_01_08_2023.sql_' from keybase
3. Run the command to access to Postgres 
 `docker-compose exec postgres psql -U postresql` 
 and create the database `CREATE DATASE wizeq`
4.  Copy the  `dump_wizeq_01_08_2023.sq` to the postgres container 
`docker cp  <REPLACE WITH YOUR PATH>/dump_wizeq_01_08_2023.sql postgres:/dump_wizeq_01_08_2023.sql
`
5. Verify that the file is inside the container 
`docker-compose exec postgres ls`

<img width="723" alt="Screen Shot 2023-08-01 at 20 44 35" src="https://github.com/wizeline/wize-q-remix/assets/97203617/bd57fb07-64a0-46b3-9cae-f9b87e9bfe87">

6.  run the command `docker-compose exec postgres psql -U postgres -d wizeq -a -f dump_wizeq_01_08_2023.sql` to copy the dump in your local DB.

7. Connect to the DB ` psql -h localhost -p 5431 -U postgres -d wizeq ` [use the password: wizeq_password]
8.  Verify that you have questions in your database with the following query: `SELECT COUNT (*) FROM wizeq.questions;`
9. Change the variable. `DATABASE_URL` in your .env file
`postgresql://USER:PASSWORD@localhost:PORT/DB?schema=SCHEMA`
You can send me a DM to share with you the values. 
11.  Run the project `npm run dev`
--- TESTING ---
1. Verify that the container for testing '**_integration-tests-prisma_**' is running `docker ps `
2. run the command `npm run test:integration` 
3. 
<img width="995" alt="Screen Shot 2023-08-03 at 12 03 06" src="https://github.com/wizeline/wize-q-remix/assets/97203617/e9a80f0d-b05e-427c-853a-358a085f6cf1">


# 👁️ 
if you want to access to the DB in Postgres use ` psql -h localhost -p 5431 -U postgres -d wizeq `


# ⚠️ 
testing job is falling because is setting for MySQL.
